dbt/adapters/base/relation.py

```diff
@@ -36,6 +37,13 @@
 SerializableIterable = Union[Tuple, FrozenSet]
 
 
+@dataclass(frozen=True, eq=False, repr=False)
+class EventTimeFilter(FakeAPIObject, Hashable):
+    field_name: str
+    start: Optional[datetime] = None
```

Comment:
do we need to do any validation on start/end time?

Reply:
Beyond them being timestamps, I think we're good 👍 The core part of the framework considers the configured partition grain and plumbs the appropriate start/end bounds while constructing an `EventTimeFilter` to initialize a `Relation` with.